### PR TITLE
Update helm chart - Include securityContext, gracePeriod and remove duplicates in helm

### PR DESCRIPTION
**Pull request by ChethanUK**
Update helm chart - Include security context, gracePeriod, etc.. and remove duplicates in helm

- Add `terminationGracePeriodSeconds` to broker, controller, minion, and server sts
- Use pinot `serviceAccountName`
- Allow custom `securityContext`
- Add `imagePullSecrets`
- Add container `securityContext`
- Add separate `livenessProbe` and `readinessProbe` for broker, controller, minion, and server 
- Allow creation of Internal GCP Load balancer for services
- etc.. and other refractoring changes and reduce duplication

### CODE REVIEW

File: kubernetes/helm/pinot/templates/broker/service-external.yaml [hunk 1]
```diff
@@ -23,19 +23,17 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ template "pinot.broker.external" . }}
+  {{- if .Values.broker.service.gcpInternalLB }}
+  annotations:
+    cloud.google.com/load-balancer-type: Internal
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service-external.yaml:28]: why make this to gcp?

Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service-external.yaml:28]: if check should go into annotations?

File: kubernetes/helm/pinot/templates/broker/service-headless.yaml [hunk 2]
```diff
@@ -22,17 +22,12 @@ kind: Service
...
   labels:
-    app: {{ include "pinot.name" . }}
-    chart: {{ include "pinot.chart" . }}
-    component: {{ .Values.broker.name }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    {{- include "pinot.brokerLabels" . | nindent 4 }}
 spec:
-  clusterIP: None
+  clusterIP: {{ .Values.broker.service.clusterIP }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service-headless.yaml:27]: I think this is the convention for headless services: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

File: kubernetes/helm/pinot/templates/broker/service-external.yaml (hunk 1 above)
Comment (xiangfu0) [reply kubernetes/helm/pinot/templates/broker/service-external.yaml:28]: annotations:
    {{- if .Values.broker.service.gcpInternalLB }}
    cloud.google.com/load-balancer-type: Internal
    {{- end. }}

File: kubernetes/helm/pinot/templates/broker/service-headless.yaml (hunk 2 above)
Comment (ChethanUK) [reply kubernetes/helm/pinot/templates/broker/service-headless.yaml:27]: Copy-paste err, Updated

File: kubernetes/helm/pinot/templates/broker/service-headless.yaml [hunk 3]
```diff
@@ -22,17 +22,12 @@ kind: Service
...
-    component: {{ .Values.broker.name }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    {{- include "pinot.brokerLabels" . | nindent 4 }}
 spec:
   clusterIP: None
   ports:
     # [pod_name].[service_name].[namespace].svc.cluster.local
-    - port: {{ .Values.broker.service.port }}
+    - name: {{ .Values.broker.service.name }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service-headless.yaml:30]: just keep port here. Same for other headless services

File: kubernetes/helm/pinot/templates/broker/service-external.yaml [hunk 4]
```diff
@@ -23,19 +23,18 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ template "pinot.broker.external" . }}
+  annotations:
+    imageTag: {{ .Values.image.tag }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service-external.yaml:27]: Why we put this imageTag for Services?

File: kubernetes/helm/pinot/templates/broker/service.yaml [hunk 5]
```diff
@@ -21,18 +21,18 @@ apiVersion: v1
...
   labels:
-    app: {{ include "pinot.name" . }}
-    chart: {{ include "pinot.chart" . }}
-    component: {{ .Values.broker.name }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    {{- include "pinot.brokerLabels" . | nindent 4 }}
 spec:
-  type: ClusterIP
+  type: {{ .Values.broker.service.type }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service.yaml:32]: what's the possible other values for this `broker.service.type`?

File: kubernetes/helm/pinot/templates/broker/service.yaml [hunk 6]
```diff
@@ -21,18 +21,18 @@ apiVersion: v1
...
   ports:
     # [pod_name].[service_name].[namespace].svc.cluster.local
-    - port: {{ .Values.broker.service.port }}
+    - name: {{ .Values.broker.service.name }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/broker/service.yaml:35]: same here, why we need this?

File: kubernetes/helm/pinot/templates/controller/service.yaml [hunk 7]
```diff
@@ -21,18 +21,18 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ include "pinot.controller.fullname" . }}
+  annotations:
+    type: "service"
+  {{- if .Values.broker.service.gcpInternalLB }}
```
Comment (xiangfu0) [inline kubernetes/helm/pinot/templates/controller/service.yaml:26]: if this indentation correct?

**Review by xiangfu0 — APPROVED** [review]

File: kubernetes/helm/pinot/templates/broker/service.yaml [hunk 8]
```diff
@@ -21,18 +21,18 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ include "pinot.broker.fullname" . }}
+  annotations:
+    type: "service"
+  {{- if .Values.broker.service.gcpInternalLB }}
+    cloud.google.com/load-balancer-type: Internal
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/templates/broker/service.yaml:27]: @ChethanUK I know the PR is merged but do you think we could improve this by removing hardcoded cloud specific annotations in favour of a fully generic annotation map? Most helm chart supports that with something like this:

```
      annotations:     
      {{- with .Values.service.annotations }}
      {{- toYaml . | nindent 8 }}
      {{- end }}
```

That would extend the support LB support to all cloud providers.

Comment (ChethanUK) [reply kubernetes/helm/pinot/templates/broker/service.yaml:27]: Ya, I was thinking more in sense of load balancer point of view :(
Sent a PR: https://github.com/apache/incubator-pinot/pull/6985

**Pull request merged**